lib/spack/docs/packages_yaml.rst

```diff
@@ -487,6 +487,42 @@ present. For instance with a configuration like:
 
 you will use ``mvapich2~cuda %gcc`` as an ``mpi`` provider.
 
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Conflicts and strong preferences
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The semantic of requirements allows for expressing both "strong preferences" and "conflicts"
+from configuration files. The syntax for doing so, though, may not be immediately clear. For
+instance, if we want to prevent any package from using ``%clang``, we can set:
+
+.. code-block:: yaml
+
+   packages:
+     all:
+       require:
+       - one_of: ['%clang', '@:']
+
+Since only one of the requirements must hold, and ``@:`` is always true, the rule above is
+equivalent to a conflict. For "strong preferences" we need to substitute the ``one_of`` policy
+with ``any_of``.
+
+Since the syntax above obfuscates the meaning of the rules added by a user, there is some
+syntactic sugar that can be used in configuration:
```

Comment:
we should update the docs after this to use some much more intuitive motivation.  e.g., "you might want to express a conflict or a strong preference".  I don't think anyone cares about the `one_of` / `any_of` equivalents, or at the very least we shouldn't put them first.

Reply:
Done in https://github.com/spack/spack/pull/41832/commits/c575b83c979ae73b876f86bfee4008d26af47d8b